.github/workflows/build-strict.yml

```diff
@@ -45,5 +41,73 @@ jobs:
           rm -rf .lake/packages/proofwidgets/.lake/build/ir
           lake build ProofWidgets
 
+      - name: confirm all modules are included
+        run: lake exe mk_all --check
+
       - name: build project
         run: lake build --wfail
+
+      - name: Build blueprint and copy to `docs/blueprint`
+        uses: xu-cheng/texlive-action@v2
+        with:
+          docker_image: ghcr.io/xu-cheng/texlive-full:20231201
+          run: |
+            apk update
+            apk add --update make py3-pip git pkgconfig graphviz graphviz-dev gcc musl-dev
+            git config --global --add safe.directory $GITHUB_WORKSPACE
+            git config --global --add safe.directory `pwd`
+            python3 -m venv env
+            source env/bin/activate
+            pip install --upgrade pip requests wheel
+            pip install pygraphviz --global-option=build_ext --global-option="-L/usr/lib/graphviz/" --global-option="-R/usr/lib/graphviz/"
+            pip install leanblueprint
+            leanblueprint pdf
+            mkdir docs
+            cp blueprint/print/print.pdf docs/blueprint.pdf
+            leanblueprint web
+            cp -r blueprint/web docs/blueprint
+
+      - name: check blueprint declarations
+        run: lake exe checkdecls blueprint/lean_decls
+
+      - name: cache mathlib docs
+        uses: actions/cache@v4
+        with:
+          path: |
+            .lake/build/doc/init
+            .lake/build/doc/lake
+            .lake/build/doc/lean
+            .lake/build/doc/std
+            .lake/build/doc/mathlib
+            .lake/build/doc/declarations
+            !.lake/build/doc/declarations/declaration-data-algebrainlean*
+          key: mathlibdoc-${{ hashfiles('lake-manifest.json') }}
+          restore-keys: |
+            mathlibdoc-
+
+      - name: build documentation
+        run: lake -R -Kenv=dev build AlgebraInLean:docs
+
+      - name: move documentation to `docs/docs`
+        run: |
+          sudo chown -R runner docs
+          cp -r .lake/build/doc docs/docs
+
+      - name: make sure the cache works
+        run: |
+          mv docs/docs .lake/build/doc
+
+      - name: upload docs & blueprint artifact
+        uses: actions/upload-pages-artifact@v3
```

Comment:
Maybe this step should only run if it's on the master branch? Not sure.

Reply:
That would require moving it into a separate job, which isn't worth the effort. An artifact is simply a file blob that's uploaded to GitHub after an action finishes; it's meant for log files and whatnot, and it's accessed in the actions tab. It's the `deploy-pages` action that uses the uploaded artifact to edit Pages.

In essence, including the artifact allows us to access the docs that were built. If they're built anyway, I don't see why we shouldn't upload them for our viewing pleasure. Otherwise they get deleted when the action is concluded.

Reply:
oh then definitely keep this as an alternative for previewing the blueprint for pending PRs.